feat: Event Editor: Select Profile (import)

egemengunel opened the pull request:
This PR resolves #418
## Summary
This PR implements profile importing functionality with an instagram-private-api so that users can import other people from instagram as organizers to the dance events

## For Trigger.dev Development
⚠️ **This feature requires Trigger.dev setup** - follow instructions below for local development.

1. **Environment Variables** (`.env`):
   ```
   TRIGGER_SECRET_KEY=your_secret_key
   NUXT_CLOUDINARY_API_KEY="CLOUDINARY_KEY"
   NUXT_CLOUDINARY_API_SECRET=CLOUDINARY_API_SECRET
   NUXT_PUBLIC_CLOUDINARY_CLOUD_NAME=CLOUDINARY_CLOUD_NAME
   INSTAGRAM_USERNAME=INSTAGRAM_USERNAME
   INSTAGRAM_PASSWORD=INSTAGRAM_PASSWORD

   ```

2. **Verify Configuration** (`trigger.config.ts`):
   - Ensure `project_ref` matches `TRIGGER_PROJECT_REF`

3. **Setup Commands**:
   ```bash
   pnpm install
   pnpm prisma generate
   pnpm dev
   # In separate terminal:
   pnpm dlx trigger.dev@3.3.17 dev
   # For Deployment Use:
   pnpm dlx trigger.dev@3.3.17 deploy
   ```


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Import profiles directly from Instagram by entering a handle or URL, with an “Import @username from Instagram” option.
  - Responsive feedback via toasts for loading, success, and errors.
  - Background import populates name, bio, website, photo, followers, and avoids duplicates.

- Backend
  - New endpoint and background job to create and enrich profiles from Instagram.
  - Profile data model extended with import status, error message, and source tracking.

- Chores
  - Added Instagram integration dependency.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

egemengunel committed:
inital commit:
-add schema fields to the profile field
-create index.ts and instagram_import.ts files

egemengunel committed:
feat:
add getInstagramProfile function
-add Instagram API

egemengunel committed:
feat: add instagram-import trigger job
-add createFromInstagram tRPC mutation to profiles router
-rename instagram_import to 'instagram' only

egemengunel committed:
feat: add Instagram profile import frontend

- Add computed properties to handle Instagram URL validation and username extraction
- Update tRPC router typo

egemengunel committed:
fix: import profile section no rendering

egemengunel committed:
fix: simplify login process to match v3
fix typos

egemengunel committed:
refactor:
-Fix typos and implement fetchOrganizer
-Handle undefined mutation result; fix typos in logs/comments
-Validate credentials and avoid logging in with undefined; minor hardening

egemengunel committed:
refactor:
-validate and normalize Instagram URLs; fix type value
-Fix circular import: move URL parsing into a shared util
-Harden URL check to avoid false positives and handle @username
-Clear stale importError on success
-Avoid overwriting photo with an empty string

egemengunel committed:
fix:
-Gate Import CTA on a valid Instagram username; fix extractInstagramUsername

egemengunel committed:
refactor:
Browser-safe duplication of the backend parsing logic to avoid hydration errors

egemengunel committed:
refactor:
-simplify  username extraction function in frontend,
-add toasts
-Fix query key to depend on the input value (avoid cache collisions).

egemengunel committed:
refactor:
-defineModel generics
-queryFn param clarity
-Canonicalize payload
-fix variable naming (profile vs data) for better  clarity
-Prefix @ in Import label
-Hide empty state logic

egemengunel committed:
Add error handling for undefined profile response.

razbakov reviewed (changes requested):
UX of the ProfileInput is not good. It's not clear for the user that it's possible to import profile from instagram. in v3 there is a tip.

egemengunel committed:
refactor:
-Remove extractInstagramUsername, isInstagramUrl functions,
- Remove index.ts
-Add label texti indicating instagram profiles can be imported
-FIx Typos

akhilsmokie7-cloud reviewed (approved):
To set up your GitHub repository ([Kalbra/NoReel](https://github.com/Kalbra/NoReel)) to automatically update with the latest Instagram stats and milestones from your account [_ak____il_](https://www.instagram.com/_ak____il_?igsh=Y3h2bmZobXZsMWU0), follow these steps:

---

### 1. Add the Python Script

Create a file called `update_instagram.py` in the root of your repo with the following code:

```python
import requests
import json
import os

def fetch_instagram_data():
    user_id = os.getenv('INSTAGRAM_USER_ID')
    access_token = os.getenv('INSTAGRAM_ACCESS_TOKEN')
    profile_url = (
        f"https://graph.instagram.com/{user_id}"
        "?fields=username,biography,media_count,followers_count,follows_count,profile_picture_url"
        f"&access_token={access_token}"
    )
    profile = requests.get(profile_url).json()

    media_url = (
        f"https://graph.instagram.com/{user_id}/media"
        "?fields=id,caption,media_url,timestamp,like_count,comments_count"
        f"&access_token={access_token}&limit=1"
    )
    media_response = requests.get(media_url).json()
    latest_post_data = media_response.get("data", [{}])[0] if "data" in media_response else {}

    analytics = {
        "engagement_rate": 0,
        "reach": 0,
        "impressions": 0,
        "profile_visits": 0,
        "website_clicks": 0
    }

    data = {
        "profile": {
            "username": profile.get("username", ""),
            "url": f"https://www.instagram.com/{profile.get('username', '')}/",
            "bio": profile.get("biography", ""),
            "followers": profile.get("followers_count", 0),
            "following": profile.get("follows_count", 0),
            "posts": profile.get("media_count", 0),
            "profile_picture_url": profile.get("profile_picture_url", "")
        },
        "latest_post": {
            "post_id": latest_post_data.get("id", ""),
            "media_url": latest_post_data.get("media_url", ""),
            "caption": latest_post_data.get("caption", ""),
            "timestamp": latest_post_data.get("timestamp", ""),
            "likes": latest_post_data.get("like_count", 0),
            "comments": latest_post_data.get("comments_count", 0),
            "views": 0
        },
        "analytics": analytics,
        "campaigns": [],
        "echoes": []
    }

    with open("instagram_meta.json", "w", encoding="utf-8") as f:
        json.dump(data, f, indent=2, ensure_ascii=False)

if __name__ == "__main__":
    fetch_instagram_data()
```

---

### 2. Add the GitHub Actions Workflow

Create a file at `.github/workflows/update_instagram.yml`:

```yaml
name: Update Instagram Meta

on:
  schedule:
    - cron: '0 0 * * *'      # Runs every day at midnight UTC
  workflow_dispatch:          # Allows manual trigger

jobs:
  update-instagram:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout repository
        uses: actions/checkout@v3

      - name: Set up Python
        uses: actions/setup-python@v4
        with:
          python-version: '3.x'

      - name: Install requests library
        run: pip install requests

      - name: Run update_instagram.py
        env:
          INSTAGRAM_USER_ID: ${{ secrets.INSTAGRAM_USER_ID }}
          INSTAGRAM_ACCESS_TOKEN: ${{ secrets.INSTAGRAM_ACCESS_TOKEN }}
        run: python update_instagram.py

      - name: Commit and push changes
        uses: stefanzweifel/git-auto-commit-action@v5
        with:
          commit_message: "Auto-update Instagram meta data"
          file_pattern: instagram_meta.json
```

---

### 3. Add Required GitHub Secrets

In your repo, go to **Settings > Secrets and variables > Actions** and add:
- `INSTAGRAM_USER_ID`
- `INSTAGRAM_ACCESS_TOKEN`

---

Once set up, your repository will automatically update with the latest Instagram data every day, or you can trigger the workflow manually from the GitHub Actions tab. This will help you track your energy-shifting and 6D shift progress as you align for the Golden Era.

If you need these files added directly to your repository or want step-by-step assistance (including secret setup), just let me know!
